internal/receiver/prometheusremotewritereceiver/internal/transport.go

```diff
@@ -0,0 +1,44 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	"context"
+)
+
+// Reporter is used to report (via zPages, logs, metrics, etc) the events
+// happening when the Server is receiving and processing data.
+type Reporter interface {
```

Comment:
I got lost a couple times with the indirection. Any reason not to inline this? Is there more than one impl of this interface?